src/lib.rs

```diff
@@ -145,15 +146,45 @@ fn get_repo() -> Result<Repository, Error> {
             "key \"workspace_root\" not found in cargo metadata".to_owned(),
         ))?
         .as_str();
-    Repository::open(primary_package_dir).map_err(|error| {
+
+    let git_root = find_valid_git_root_inner().map_err(|error| {
+        Error::new(
+            Span::call_site(),
+            format!("failed to find Git Repo: {error}"),
+        )
+    })?;
+
+    Repository::open(git_root).map_err(|error| {
         Error::new(
             Span::call_site(),
-            format!("failed to open repository at location {primary_package_dir}: {error}"),
+            format!("failed to find Git Repo: {error}"),
```

Comment:
Can we kinda revert this? Would be useful to see the different error message if it happens:
```suggestion
            format!("failed to open repository at {git_root}: {error}"),
```